.mergify.yml

```diff
@@ -0,0 +1,16 @@
+pull_request_rules:
+  - name: Automatic merge on approval
+    conditions:
+      - base=master
+      - status-success="Run test suite"
+      - status-success="validate commits"
+      - label="merge-when-passing"
+      - label!="work-in-progress"
+      - "#approved-reviews-by>=1"
+      - "#changes-requested-reviews-by=0"
+      - -title~=^\[*[Ww][Ii][Pp]
+    actions:
+      merge:
+        method: rebase
+        rebase_fallback: merge
```

Comment:
Same question as above?

Reply:
A contributor having a merge commit in their PR is _technically_ allowed by the [current proposed contributing guide](https://github.com/ExaWorks/SDK/pull/25/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R30).  In that scenario, I see three options:

1. We fallback to merging (while still encouraging regular contributors to use the rebase workflow so that merging is rare)
3. We modify the contributing guide to require rebases/disallow merges and force the contributor to rebase their PR, dropping any merge commits (a daunting task for anyone unfamiliar with git and rebasing - so we may lose contributions)
2. We have the maintainers take on the responsibility of resolving conflicts during a rebase (seems like the worst option)

Either way, this seems like maybe a conversation to have over in #25, and then we codify the outcome of that here.  I'll make a comment in #25 referencing this.